packages/controllers/src/snaps/SnapController.ts

```diff
@@ -367,6 +368,8 @@ type SnapControllerMessenger = RestrictedControllerMessenger<
   AllowedEvents['type']
 >;
 
+type GetAppKey = (subject: string) => Promise<string>;
```

Comment:
I'd prefer we wouldn't prepend `encryption:` to the Snap Id and probably do brittle string operations in the extension.
How about
```typescript
enum AppKeyTypes {
  stateEncryption
}
type GetAppKey = (subject: string, type: AppKeyTypes) => Promise<string>;
```